rasa/nlu/classifiers/diet_classifier.py

```diff
@@ -657,19 +773,35 @@ def _use_default_label_features(self, label_ids: np.ndarray) -> List[FeatureArra
 
     def _create_model_data(
         self,
-        training_data: List[Message],
-        label_id_dict: Optional[Dict[Text, int]] = None,
-        label_attribute: Optional[Text] = None,
+        messages: List[Message],
+        *,
         training: bool = True,
+        label_id_dict: Optional[Dict[Text, int]] = None,
     ) -> RasaModelData:
-        """Prepare data for training and create a RasaModelData object."""
+        """Prepare data for training and create a RasaModelData object.
+
+        Args:
+          messages: the list of messages to be turned into a RasaModelData
+          training: whether we're in training mode and should add label data
+          label_id_dict: a mappig from labels (Text) to label ids (ints) that is
+            only required for label data creation (i.e. if `training` is set to `True`)
```

Comment:
Yes we could (first step is checking whether label_id_dict is set and it's only used when training is True) - but I think it's good to keep this explicit for users as a reminder that "this is for training - and oh if you want to use this for training please pass ..." instead of going "oh, there's a label dict, well then I switch to training mode"